documentation/architecture/ADR1:dependency-injection.md

```diff
@@ -0,0 +1,76 @@
+## Context
+##### What is DI?
+
+In DI, client code does not create it's own dependencies.
+```kotlin
+// Without DI, Repository creates it's own PocketEndpoint
+class Repository {
+    val endpoint = PocketEndpoint()
+}
+
+class Parent {
+    val repo = Repository()
+}
+```
+```kotlin
+// With DI, endpoint is injected (passed) through the constructor...
+class Repository(val endpoint: Endpoint) {
+}
+
+class Parent {
+    val repo = Repository(PocketEndpoint())
+}
+
+// ... or through a field
+class Repository {
+    var endpoint: Endpoint
+}
+
+class Parent {
+    val repo = Repository()
+
+    init {
+        repo.endpoint = PocketEndpoint()
+    }
+}
+```
+See https://en.wikipedia.org/wiki/Dependency_injection
+
+##### Why inject?
+
+Injection makes testing simpler. When testing, we don't want our repos to make normal network calls (these make tests flaky, as calls can fail or return unexpected data). So we provide fake endpoints that return the data we need. This is much simpler with DI.
+
+It also makes it easier to follow the adage "program to an interface, not an implementation." Client code (in this case, the Repository) can define the interface upon which they depend, and then never even import the actual implementation. This helps us to separate concerns.
+
+##### Dependency injection vs service locators
+
+We use a service locator pattern. This is similar to, but distinct from dependency injection.
```

Comment:
This should be part of the `decision` section but sections after this are more appropriate for the context.

It also sounds like what we currently do, as opposed to what we will do.